src/SchemaRegistry.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.21;
+
+import { Schema } from "./struct/Schema.sol";
+
+contract SchemaRegistry {
+  mapping(bytes32 id => Schema schema) public schemas;
+  uint256 public numberOfSchemas = 0;
+
+  error SchemaAlreadyExists();
+  error SchemaNameMissing();
+  error SchemaStringMissing();
+
+  event SchemaCreated(string name, bytes32 schemaId);
+
+  function getIdFromSchemaString(string calldata _schema) public pure returns (bytes32) {
+    return keccak256(abi.encodePacked(_schema));
+  }
+
+  function createSchema(
+    string calldata name,
```

Comment:
if we use calldata here, a smart contract could not use this method; 
Better to use memory